scarlet/initialization.py

```diff
@@ -157,27 +189,71 @@ def trim_morphology(center_index, morph, bg_thresh):
                 bbox.stop[1] - center_index[-1],
             )
         )
-        while boxsize < size:
-            boxsize += 16  # keep box sizes quite small
+    else:
+        size = 0
 
-    # define bbox and trim to bbox
+    # define new box and cut morphology accordingly
+    boxsize = get_minimal_boxsize(size)
     bottom = center_index[0] - boxsize // 2
     top = center_index[0] + boxsize // 2
     left = center_index[1] - boxsize // 2
     right = center_index[1] + boxsize // 2
     bbox = Box.from_bounds((bottom, top), (left, right))
     morph = bbox.extract_from(morph)
-    bbox = Box.from_bounds((bottom, top), (left, right))
     return morph, bbox
 
 
+def init_compact_source(
+    sky_coord, frame, observations,
+):
```

Comment:
Maybe a docstring would help. I suspect users who declare new sources will want to use or at least look at initialisation functions.

Reply:
+1